src/Bicep.Core/Syntax/ModuleDeclarationSyntax.cs

```diff
@@ -1,76 +1,84 @@
-// Copyright (c) Microsoft Corporation.
-// Licensed under the MIT License.
-using System.Collections.Generic;
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System;
+using System.Collections.Generic;
 using System.Linq;
-using Bicep.Core.Navigation;
-using Bicep.Core.Parsing;
-using Bicep.Core.Semantics;
-using Bicep.Core.TypeSystem;
-
-namespace Bicep.Core.Syntax
-{
+using Bicep.Core.Navigation;
+using Bicep.Core.Parsing;
+using Bicep.Core.Semantics;
+using Bicep.Core.TypeSystem;
+
+namespace Bicep.Core.Syntax
+{
     public class ModuleDeclarationSyntax : StatementSyntax, INamedDeclarationSyntax
     {
-        public ModuleDeclarationSyntax(IEnumerable<SyntaxBase> leadingNodes, Token keyword, IdentifierSyntax name, SyntaxBase path, SyntaxBase assignment, SyntaxBase? ifCondition, SyntaxBase body)
+        public ModuleDeclarationSyntax(IEnumerable<SyntaxBase> leadingNodes, Token keyword, IdentifierSyntax name, SyntaxBase path, SyntaxBase assignment, SyntaxBase value)
             : base(leadingNodes)
-        {
-            AssertKeyword(keyword, nameof(keyword), LanguageConstants.ModuleKeyword);
-            AssertSyntaxType(name, nameof(name), typeof(IdentifierSyntax));
-            AssertSyntaxType(path, nameof(path), typeof(StringSyntax), typeof(SkippedTriviaSyntax));
-            AssertTokenType(keyword, nameof(keyword), TokenType.Identifier);
-            AssertSyntaxType(assignment, nameof(assignment), typeof(Token), typeof(SkippedTriviaSyntax));
-            AssertTokenType(assignment as Token, nameof(assignment), TokenType.Assignment);
-            AssertSyntaxType(ifCondition, nameof(ifCondition), typeof(SkippedTriviaSyntax), typeof(IfConditionSyntax));
-            AssertSyntaxType(body, nameof(body), typeof(SkippedTriviaSyntax), typeof(ObjectSyntax));
-
-            this.Keyword = keyword;
-            this.Name = name;
-            this.Path = path;
-            this.Assignment = assignment;
-            this.IfCondition = ifCondition;
-            this.Body = body;
-        }
-
-        public Token Keyword { get; }
-
-        public IdentifierSyntax Name { get; }
-
-        public SyntaxBase Path { get; }
-
-        public SyntaxBase Assignment { get; }
-
-        public SyntaxBase? IfCondition { get; }
```

Comment:
What are your thoughts on updating `grammar.md`? Technically not necessary, but I've been trying to keep it aligned with the syntax/parser structure.

Reply:
Yeah good pt. I'll push out an update.